src/dbnode/persist/fs/merger.go

```diff
@@ -128,6 +130,8 @@ func (m *merger) Merge(
 		NamespaceMetadata: nsMd,
 		Shard:             shard,
 		BlockStart:        startTime,
+		VolumeIndex:       fileID.VolumeIndex + 1,
+		FileSetType:       persist.FileSetFlushType,
 		DeleteIfExists:    false,
 	}
 	prepared, err := flushPreparer.PrepareData(prepareOpts)
```

Comment:
Hm, I can't seem to make a comment here - but just realized on line 195:
```
idsToFinalize = append(idsToFinalize, id)
if err != nil {
  return err
}
```

The appending of `idsToFinalize` needs to happen after the error check, otherwise you'll append `nil` to the slice, and then when the defer tries to finalize each ID it will panic on nil pointer for `id.Finalize()`.

Reply:
Similarly `tagsToFinalize = append(tagsToFinalize, tags)` should come after the error check.

Reply:
Good catch 👍